src/DataTransferObject.php

```diff
@@ -66,12 +64,37 @@ protected function boot(array $parameters): void
         }
 
         $this->processRemainingProperties($parameters);
+        $this->determineImmutability();
     }
 
-    protected function resolveImmutable()
+    protected function determineImmutability()
     {
         if ($this instanceof immutable) {
```

Comment:
It's still unclear to me why we need this interface, can you explain?

Reply:
If a user wants to make a dto immutable on runtime he can do it by implementing the immutable interface. This is a nice addition for people who want immutable by default.

Reply:
```
class MyDto extends DataTransferObject implements immutable
{
    /** @var string */
    public $name;
}
```

Reply:
Ok, I'd rename `immutable` to `Immutable` though

Reply:
Sure.